test/functional/forall/test-forall-rangesegment.hpp

```diff
@@ -0,0 +1,76 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef __TEST_FORALL_RANGESEGMENT_HPP__
+#define __TEST_FORALL_RANGESEGMENT_HPP__
+
+#include "RAJA/RAJA.hpp"
+#include "camp/resource.hpp"
+
+#include "test-forall.hpp"
+
+using namespace camp::resources;
+using namespace RAJA;
+using namespace camp;
+
+template<typename INDEX_TYPE, typename WORKING_RES, typename EXEC_POLICY>
+void ForallRangeSegmentFunctionalTest(INDEX_TYPE first, INDEX_TYPE last)
+{
+  RAJA::TypedRangeSegment<INDEX_TYPE> r1(first, last);
+  INDEX_TYPE N = r1.end() - r1.begin();
+
+  camp::resources::Resource working_res{WORKING_RES()};
+  camp::resources::Resource check_res{Host()};
+
+  INDEX_TYPE * working_array = working_res.allocate<INDEX_TYPE>(N);
+  INDEX_TYPE * check_array   = check_res.allocate<INDEX_TYPE>(N);
+  INDEX_TYPE * test_array    = check_res.allocate<INDEX_TYPE>(N);
+
+  for(INDEX_TYPE i=0; i < N; i++)
```

Comment:
For brevity, you could change this loop to std::iota( test_array, test_array + N, *r1.begin() );

Now that I read this, it doesn't look much shorter :-)